Sources/tart/Commands/Run.swift

```diff
@@ -112,7 +112,19 @@ struct Run: AsyncParsableCommand {
 
   @MainActor
   func run() async throws {
-    let vmDir = try VMStorageLocal().open(name)
+    let localStorage = VMStorageLocal()
+    let vmDir = try localStorage.open(name)
+    let needToGenerateNewMac = try localStorage.list().contains {
+      // check if there is a running VM with the same MAC but different name
+      try $1.running() && $1.macAddress() == vmDir.macAddress() && $1.name != vmDir.name
+    }
+
+    // generate new MAC only after locking to make sure it's not this VM which is running
```

Comment:
The comment seems to be misleading: there's no locking done here.

We should probably at least take a `TART_HOME` lock when checking for VMs with duplicate MAC-addresses.

Reply:
Oh. This comment was left from before when I tried generating in a different place. `needToGenerateNewMac` actually filters out by name as well so we should be good.